tests/invalid_models_tests/test_models.py

```diff
@@ -761,3 +761,69 @@ class ParkingLot(Place):
             'as an implicit link is deprecated.'
         )
         self.assertEqual(ParkingLot._meta.pk.name, 'parent')
+
+    def test_m2m_table_name_clash(self):
+        class Foo(models.Model):
+            bar = models.ManyToManyField('Bar', db_table='myapp_bar')
+
+            class Meta:
+                db_table = 'myapp_foo'
+
+        class Bar(models.Model):
+            class Meta:
+                db_table = 'myapp_bar'
+
+        errors = Foo.check()
+        expected = [
+            Error(
+                "Field is using a table that has already been registered by "
+                "'invalid_models_tests.Bar.bar'.",
+                obj=Foo._meta.get_field('bar'),
+                id='fields.E340',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_m2m_field_table_name_clash(self):
+        class Foo(models.Model):
+            pass
+
+        class Bar(models.Model):
+            foos = models.ManyToManyField(Foo, db_table='clash')
+
+        class Baz(models.Model):
+            foos = models.ManyToManyField(Foo, db_table='clash')
+
+        errors = Bar.check()
+        expected = [
+            Error(
+                "Field is using a table that has already been registered by "
+                "'invalid_models_tests.Baz_foos.foos'.",
```

Comment:
As the `db_table` is attached to an `auto_created` model the message should be `Field is using a table that has already been registered by 'invalid_models_tests.Baz.foos'.`. The intermediary model `Baz_foos` is an implementation detail and shouldn't be referred to, plus it has no `foos` field.